modules/SystemTest/src/server/routes/SystemTestRoutes.coffee

```diff
@@ -86,44 +147,102 @@ exports.getOrCreateACASBobInternal = (callback) ->
 				callback statusCode, {
 					hasError: hasError
 					messages: body
-					created: created
 				}
 
-exports.deleteACASBob = (req, resp) ->
-	exports.deleteACASBobInternal (statusCode, output) ->
-		resp.statusCode = statusCode
-		resp.json output
-
-exports.deleteACASBobInternal = (callback) ->
-	exports.getACASBobUser (bob) ->
-		if !bob?
-			callback 400, {
-				hasError: true
-				messages: bob
-			}
-		else
-			options =
-				method: 'DELETE'
-				url: "#{config.all.client.service.persistence.fullpath}authors/#{bob.id}"
-				json: true
-			request options, (error, response, body) ->
+add_project_roles = (projectNames, roles, username, callback) ->
+	if !projectNames?
+		projectNames = []
+	# Look up projects and translate names to codes
+	options =
+		method: 'GET'
+		json: true
+		url: "#{config.all.server.nodeapi.path}/api/projects/getAllProjects/stubs"
+	request options, (error, response, body) ->
+		if error
+			throw new Error(error)
+		_.each projectNames, (projectName) ->
+			project = _.findWhere body, {"name":projectName}
+			if project?
+				roles.push {
+					roleType: 'Project'
+					roleKind: project.code
+					roleName: 'User'
+					userName: username
+				}
+		callback roles
+
+# Grant roles to a test user
+exports.giveTestUserRolesInternal = (username, acas_user, acas_admin, cmpdreg_user, cmpdreg_admin, projects, callback) ->
```

Comment:
This doesn't handle all the possible roles but it covers the main set, and I thought this was an easier API than having to form all these author role objects outside of ACAS.